Extension/src/LanguageServer/extension.ts

```diff
@@ -82,6 +99,113 @@ export function activate(activationEventOccurred: boolean): void {
+        const languageAssociationFilter: (info: configs.CompilerInfo) => boolean = (info: configs.CompilerInfo): boolean => {
+            return (info.languageAssociation === 'cpp' && activeFileIsCpp) || (info.languageAssociation === 'c' && activeFileIsC);
+        };
+        compilerPaths = compilerInfo.filter(languageAssociationFilter).map<string>(info => { return info.path; });
+
+        if (userCompilerPath) {
+            compilerPaths.push(userCompilerPath);
+        }
+
+        let map: Map<string, string> = new Map<string, string>();
+        compilerPaths.forEach(value => {
+            const basename: string = path.basename(value);
+            map.has(basename) ? map[basename] = value : map.set(basename, value);
+        });
+        compilerPaths = [...map.values()];
+    } else if (userCompilerPath) {
+        compilerPaths = [userCompilerPath];
+    }
+
+    if (!compilerPaths) {
+        telemetry.logLanguageServerEvent('noCompilerFound'); // Don't prompt a message yet until we can make a data-based decision
+        // Display a message prompting the user to install compilers if none were found
+        // const dontShowAgain: string = "Don't Show Again";
+        // const learnMore: string = "Learn More";
+        // const message: string = "No C/C++ compiler found on the system. Please install a C/C++ compiler to use the C/Cpp: build active file tasks.";
+
+        // let showNoCompilerFoundMessage: PersistentState<boolean> = new PersistentState<boolean>("CPP.showNoCompilerFoundMessage", true);
+        // if (showNoCompilerFoundMessage) {
+        //     vscode.window.showInformationMessage(message, learnMore, dontShowAgain).then(selection => {
+        //         switch (selection) {
+        //             case learnMore:
+        //                 const uri: vscode.Uri = vscode.Uri.parse(`https://go.microsoft.com/fwlink/?linkid=864631`);
+        //                 vscode.commands.executeCommand('vscode.open', uri);
+        //                 break;
+        //             case dontShowAgain:
+        //                 showNoCompilerFoundMessage.Value = false;
+        //                 break;
+        //             default:
+        //                 break;
+        //         }
+        //     });
+        // }
+        return [];
+    }
+
+    // The build task output file should include a '.exe' extension on Windows
+    let platformInfo: PlatformInformation = await PlatformInformation.GetPlatformInformation();
+    let exeName: string;
+    if (platformInfo.platform === 'win32') {
+        exeName = '${fileBasenameNoExtension}.exe';
+    } else {
+        exeName = '${fileBasenameNoExtension}';
+    }
+
+    // Generate tasks
+    let result: vscode.Task[] = [];
+    compilerPaths.forEach(compilerPath => {
+        const taskName: string = path.basename(compilerPath) + " build active file";
+        const args: string[] = ['-g', '${fileDirname}/${fileBasename}', '-o', '${fileDirname}/' + exeName];
```

Comment:
Also, to be easier on the user, could we do <command> -g *.<whatever their file extension choice is such as cpp> -o <filename>.<exe> ?

Reply:
Based on the file extension of the currently opened file? Or chain all of them? I don't imagine beginning users who would use this to use .c .cpp and .cxx all together in the same project.